Sources/Common/Store/FileStorage.swift

```diff
@@ -0,0 +1,137 @@
+import Foundation
+
+public struct SdkFile {
+    let type: FileType
+    let directoryPath: URL
+    let filePath: URL
+}
+
+public protocol FileStorage: AutoMockable {
+    /// return `true` if the save was successful and no errors were caught/logged
+    /// `fileId` - the file name. `nil` if
+    func save(type: FileType, contents: Data, fileId: String?) -> Bool
+    /// return `nil` if an error was caught and logged *or* if the file simply doesn't exist
+    func get(type: FileType, fileId: String?) -> Data?
+    func delete(type: FileType, fileId: String) -> Bool
```

Comment:
Wondering if `fileId` parameter for `delete` is non-optional or it has been left out mistakenly. I am asking because `save` and `get` have `fileId` as an optional parameters.